devices/Ft6xx6x/samples/Program.cs

```diff
@@ -1,55 +1,226 @@
+            nanoFramework.Console.WriteLine("Core2 Touch screen test");
+            nanoFramework.Console.WriteLine("Touch Chip type: " + Iot.Device.Ft6xx6x.ChipType.Ft6336U.ToString());
 
-void TouchInterrupCallback(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
-{
-    Debug.WriteLine("Touch interrupt");
-    var points = sensor.GetNumberPoints();
-    if (points == 1)
+            I2cConnectionSettings settings = new(1, Ft6xx6x.DefaultI2cAddress);
+            using I2cDevice device = I2cDevice.Create(settings);
+            using GpioController gpio = new();
+
+            using Ft6xx6x sensor = new(device);
+            var ver = sensor.GetVersion();
+            Debug.WriteLine($"version: {ver}");
+            sensor.SetInterruptMode(false);
+            Debug.WriteLine($"Period active: {sensor.PeriodActive}");
+            Debug.WriteLine($"Period active in monitor mode: {sensor.MonitorModePeriodActive}");
+            Debug.WriteLine($"Time to enter monitor: {sensor.MonitorModeDelaySeconds} seconds");
+            Debug.WriteLine($"Monitor mode: {sensor.MonitorModeEnabled}");
+            Debug.WriteLine($"Proximity sensing: {sensor.ProximitySensingEnabled}");
+
+            gpio.OpenPin(39, PinMode.Input);
+            // This will enable an event on GPIO39 on falling edge when the screen if touched
+            gpio.RegisterCallbackForPinValueChangedEvent(39, PinEventTypes.Falling, TouchInterrupCallback);
+
+            touchPanelUserAppTemplate touchPanelUserApp = new touchPanelUserAppTemplate();
+            TouchPanelProcessor.TouchedEvent += touchPanelUserApp.userDefinedTouchPanelBehaviourSkeleton;
+
+            while (true)
+            {
+                Thread.Sleep(20);
+            }
+
+            void TouchInterrupCallback(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
+            {
+                Debug.WriteLine("Touch interrupt");
+                var points = sensor.GetNumberPoints();
+                if (points == 1)
+                {
+                    var point = sensor.GetPoint(true);
+                    Debug.WriteLine($"ID: {point.TouchId}, X: {point.X}, Y: {point.Y}, Weight: {point.Weigth}, Misc: {point.Miscelaneous}, Event: {point.Event}");
+                    DiscrimitateButtons(point.X, point.Y);
+                }
+                else if (points == 2)
+                {
+                    var dp = sensor.GetDoublePoints();
+                    Debug.WriteLine($"ID: {dp.Point1.TouchId}, X: {dp.Point1.X}, Y: {dp.Point1.Y}, Weight: {dp.Point1.Weigth}, Misc: {dp.Point1.Miscelaneous}, Event: {dp.Point1.Event}");
+                    Debug.WriteLine($"ID: {dp.Point2.TouchId}, X: {dp.Point2.X}, Y: {dp.Point2.Y}, Weight: {dp.Point2.Weigth}, Misc: {dp.Point2.Miscelaneous}, Event: {dp.Point1.Event}");
+                }
+            }
+
+            void DiscrimitateButtons(int x, int y)
+            {
+                const double pxPerMm = 6.90415;
```

Comment:
While I understand you want to be precise, having simple square buttons will drastically reduce the calculation and speed up the process. Currently, the difference between in and outside the button is just few pixels. And with a finger, it's not that precise. I know you can use a pen, still, I would think a regular square button will just make the trick.

Reply:
it is not only a question of being precise, at the end what it counts specially for embedded systems is to have an efficient code that performs with less instructions (=uses less memory) and executes quicker....the code as it is is surely not optimized as it performs those few calculations  on each event; i will try to avoid that....in any case please, i dont want to redo the algorithm at this stage, it is taking me too much time just for the first PR to be accepted...if i also have to rework the algorithmic part it is too much...there will be always time later to refactor and optimize, dont you agree?

Reply:
I understand the frustration! Really! This way of the round buttons is not blocking. It can be kept. No problem. Leave it.